rnote-ui/src/penssidebar/shaperpage.rs

```diff
@@ -386,3 +340,55 @@ impl ShaperPage {
         );
     }
 }
+
+fn shape_builder_type_icons_get_groups() -> Vec<GroupedIconPickerGroupData> {
```

Comment:
why not use a static?

Reply:
That doesn't seem to work because `StringList` doesn't implement Send/Sync, unless I'm missing something:

```
`*mut c_void` cannot be sent between threads safely
the trait `std::marker::Send` is not implemented for `*mut c_void`
required for `TypedObjectRef<*mut c_void, ()>` to implement `Sync`
[...]
main.rs(1, 1): required because it appears within the type `gtk4::StringList`
```

`GroupedIconPickerGroupData` could use e.g. `Vec<String>` instead of `StringList` (and `GroupedIconPicker#set_groups()` could construct the `StringList`s on the fly), but I'm not sure if that's better. 
Also, the group names should get re-evaluated every time because they use `gettext`, right?

Reply:
https://docs.rs/once_cell/latest/once_cell/unsync/struct.Lazy.html maybe? I think the vec's values get written only once on first use

Reply:
I can't get that to work either; you have to make it `const` (`static` throws the same error that I posted above), but if you do that, the data is moved after accessing it

Reply:
okay, I'm not sure what the issue is then. In the end it doesn't really matter anyway because the setup is only called once but I personally like making it Vec<String> and still use lazy. But it's up to you

Reply:
I'm also fine with either option, but I'd rather leave it as-is :P

If we store the icon names as `Vec<String>`, then we'd have to iterate all elements in order to construct the `StringList`: `StringList::new(&icon_names.iter().map(String::as_str).collect::<Vec<&str>>())`

Reply:
ok!